pkg/pillar/cmd/domainmgr/metric.go

```diff
@@ -59,9 +60,52 @@ func getAndPublishMetrics(ctx *domainContext, hyper hypervisor.Hypervisor) {
 	}
 
 	hm, _ := hyper.GetHostCPUMem()
+	if hyper.Name() != "xen" {
+		// the the hypervisor other than Xen, we don't have the Dom0 stats. Get the host
+		// cpu and memory for the device here
+		formatAndPublishHostCPUMem(ctx, hm)
+	}
 	ctx.pubHostMemory.Publish("global", hm)
 }
 
+func formatAndPublishHostCPUMem(ctx *domainContext, hm types.HostMemory) {
+	var hostUUID types.UUIDandVersion
+	var usedPerc float64
+	var busy int
```

Comment:
You're casting this to a uint64 at the end, and the input is a float64. Probably makes sense to keep it as a float64 and do the truncation to uint64 at the end.